pkg/bgpv1/manager/store/diffstore.go

```diff
@@ -115,22 +127,36 @@ func (ds *diffStore[T]) handleEvent(event resource.Event[T]) {
 	event.Done(nil)
 }
 
-// Diff returns a list of items that have been upserted(updated or inserted) and deleted since the last call to Diff.
-func (ds *diffStore[T]) Diff() (upserted []T, deleted []resource.Key, err error) {
+// InitDiff initializes tracking io items to Diff for the given callerID.
+func (ds *diffStore[T]) InitDiff(callerID string) {
+	ds.mu.Lock()
+	defer ds.mu.Unlock()
+
+	ds.callerUpdatedKeys[callerID] = make(updatedKeysMap)
+}
+
+// Diff returns a list of items that have been upserted (updated or inserted) and deleted since the last call
+// to Diff with the given callerID.
+func (ds *diffStore[T]) Diff(callerID string) (upserted []T, deleted []resource.Key, err error) {
 	ds.mu.Lock()
 	defer ds.mu.Unlock()
 
 	if ds.store == nil {
 		return nil, nil, ErrStoreUninitialized
 	}
 
+	updatedKeys, ok := ds.callerUpdatedKeys[callerID]
+	if !ok {
+		return nil, nil, fmt.Errorf("diff not initialized for caller %s", callerID)
```

Comment:
It would be nice to have defined error type for this condition, which caller can use to resolve it.

Reply:
defined an error for it